document/CoreFormat.mdk

```diff
@@ -2439,9 +2440,35 @@ C++: `"std::runtime_error"`
 C#: `"System.ArgumentNullException"`
 ~
 
+If the tool does not have access to information about the object that was thrown, the `type` property shall be absent.
+
+~ IsoNote
+This will be the case in C++ programs that throw objects that do not derive from `std::exception`:
```

Comment:
So your guidance for C++ is "populate message from what()" (which guidance I do have below), "... and don't bother with type".